### PR TITLE
feat: creating a state to keep index config data for each entity #272

### PR DESCRIPTION
**Pull request by BruceRodrigues**
Closes https://github.com/clevercanary/data-explorer/issues/272

### CODE REVIEW

File: packages/data-explorer-ui/src/providers/exploreState.tsx [hunk 1]
```diff
@@ -330,6 +357,14 @@ type UpdateSortingAction = {
   type: ExploreActionKind.UpdateSorting;
 };
 
+/**
+ * Update column visibility action.
+ */
+type UpdateColumnVisibilityAction = {
+  payload: Record<string, boolean>;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/providers/exploreState.tsx:364]: Could we use react table `VisibilityState`?

File: packages/data-explorer-ui/src/providers/exploreState.tsx [hunk 2]
```diff
@@ -203,6 +219,16 @@ export function ExploreStateProvider({
       exploreReducer(s, a, { config, entityConfig }),
     {
       categoryViews: [],
+      entityPageState: config.entities.reduce(
+        (acc, { list: { columns }, route }) => ({
+          ...acc,
+          [route]: {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/providers/exploreState.tsx:225]: I wonder if we can get some reuse out of https://github.com/clevercanary/data-explorer/blob/775f22cc75643348a99f752de181f94a154cb640/packages/data-explorer-ui/src/components/Table/common/utils.ts#L247?

If we grab `defaultSort` from line `223` e.g.  update `223` with:

```(acc, { list: { columns, defaultSort }, route }) => ({```

and then update lines `225-228` with:

```
[route]: getInitialState(columns, defaultSort)
```

Let me know if this is suitable re-use?

File: packages/data-explorer-ui/src/providers/exploreState.tsx [hunk 3]
```diff
@@ -69,11 +70,27 @@ export interface ExploreStaticResponse {
   paginationResponse: PaginationResponse;
 }
 
+/**
+ * State for each entity
+ */
+export interface EntityPageState {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/providers/exploreState.tsx:76]: Should we update line `85` to use either `InitialTableState` or `TableState` from react table?

.e.g.

```
/**
 * State for all entities
 */
export interface EntityPageStateMapper {
  [key: EntityPath]: TableState;
}
```

**Review by frano-m — APPROVED** [review]
⭐ @BruceRodrigues thank you!

**Comment by NoopDog** [comment]
Thanks! @BruceRodrigues and @frano-m ⭐ ⭐ !!

**Pull request merged**